core/ecschema-editing/src/Editing/ChangeInfo/ChangeInfo.ts

```diff
@@ -0,0 +1,151 @@
+import { SchemaItemType } from "@itwin/ecschema-metadata";
+import { SchemaContextEditor } from "../Editor";
+import { ChangeOptionProps, ChangeOptions } from "./ChangeOptions";
+import { SchemaEditType } from "../SchemaEditType";
+
+/**
+ * Defines the function used to revert a specific edit defined by the ISchemaEditChangeInfo type parameter.
+ * @alpha
+ */
+export type SchemaChangeRevertCallback = <T extends ISchemaEditChangeInfo>(changeInfo: T) => Promise<void>;
+
+/**
+ * Defines the function used begin (true) or cancel (false) a specific edit operation defined by the ISchemaEditChangeInfo type parameter.
+ * @alpha
+ */
+export type BeginSchemaEditCallback= <T extends ISchemaEditChangeInfo>(changeInfo: T) => Promise<boolean>;
+
+/**
+ * @alpha
+ */
+export interface ISchemaEditChangeProps {
+  readonly changeOptions?: ChangeOptionProps;
+  readonly editType?: string;
+  readonly schemaItemType?: string;
+}
+
+/**
+ * Defines the properties required for a specific schema edit operation.
+ * @alpha
+ */
+export interface ISchemaEditChangeInfo {
```

Comment:
Do we really need that interface? Since we have an abstract base class of all edits, we could also use that as carry type in the lists and references. As far as I see it, only toJson and beginChange is hidden in that interface, so we don't gain much by introducing another interface.

By the way, you can import classes acting as "interfaces" by using the type indicator:
```ts 
import type { SchemaEditChangeBase } from "./ChangeInfo"
```